### PR TITLE
add this script to package gcs connector into a Cloudera parcel can b…

### PR DESCRIPTION
**Pull request by yhqs540**
This tool is to download and create a Cloudera parcel for Google Cloud Storage connector. The parcel then can be used by Cloudera Hadoop cluster to install GCS connector and access to GCS.

### CODE REVIEW

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+# User must run the script in a separate folder where only the files to be packaged are present
+# -f: name of the parcel in a single string format without any spaces or special characters.
+# -v: version of the parcel in the format x.x.x (ex: 1.0.0)
+# -o: name of the operating system distribution
+# -d: flag is to be used if you want to deploy the parcel to the cloudera manager parcel repo
+# folder, this flag is optional and if not provided then the parcel file will be created in the
+# same directory where script run.
+
+while getopts f:v:o:d: option
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Put do on same line. https://google.github.io/styleguide/shell.xml#Loops

```sh
while getopts f:v:o:d: option; do
```

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+do
+case "${option}"
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Use indentation of 2 spaces. https://google.github.io/styleguide/shell.xml#Indentation

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+in
+f) filen=${OPTARG};;
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Add space before `;;` for this and following statements
https://google.github.io/styleguide/shell.xml#Case_statement

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+v) version=${OPTARG};;
+o) OSTYPE=${OPTARG};;
+d) placefile='true';;
+esac
+done
+
+#capture logs
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Leave a more detailed comment here.

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+>/var/log/build_parcel.log
+exec 1>/var/log/build_parcel.log 2>&1
+
+if [[ "$OSTYPE" == "el5" ]];
+	then OS=el5 ; yum install wget -y
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Place `then` on same line as if statement. 
https://google.github.io/styleguide/shell.xml#Loops

Also split the OS assignment and install to separate lines.

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+elif [[ "$OSTYPE" == "squeeze" ]];
+	then OS=squeeze ; apt-get install wget -y
+elif [[ "$OSTYPE" == "wheezy" ]];
+	then OS=wheezy ; apt-get install wget -y
+else
+     echo "OS not in list, please provide valid OS name"
+fi
+
+## Display variable information input by user
+#echo "Downloaded file:$link"
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Should this commented code be removed?

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+echo "Version number:$version"
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Prefer `${VAR}` style for this and following.
https://google.github.io/styleguide/shell.xml#Variable_expansion

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+	then OS=trusty ; apt-get install wget -y
+elif [[ "$OSTYPE" == "squeeze" ]];
+	then OS=squeeze ; apt-get install wget -y
+elif [[ "$OSTYPE" == "wheezy" ]];
+	then OS=wheezy ; apt-get install wget -y
+else
+     echo "OS not in list, please provide valid OS name"
+fi
+
+## Display variable information input by user
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Why extra `#`

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+#echo "Downloaded file:$link"
+echo "Version number:$version"
+echo "OS type:$OS"
+echo "Filename:$filen"
+mkdir -p ${filen^^}-$version/lib/hadoop/lib && mkdir -p ${filen^^}-$version/meta
+touch ${filen^^}-$version/meta/parcel.json
+
+###Download gcs connector jar and copy all folder content to parcel location
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Remove extra `#` and add space between comment char and comment.

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Remove new-line char between comment and code

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+wget https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-hadoop2-latest.jar
+
+cp * ${filen^^}-$version/lib/hadoop/lib/
+
+
+## Create parcel.json file required for parcel packaging
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Remove extra `#`a

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Remove new-line char between comment and code.

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+  ],
+
+  "users": { },
+
+  "groups": [ ]
+}
+
+EOL
+
+##export HADOOP_CLASSPATH
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Add more detailed comment or remove.

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+cat >>${filen^^}-$version/meta/$filen.sh<< EOL
+#!/bin/bash
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Is this declaration required?

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+export HADOOP_CLASSPATH=\$HADOOP_CLASSPATH:/opt/cloudera/parcels/$filen-$version/lib/hadoop/lib/gcs-connector-latest-hadoop2.jar
+EOL
+
+## Create parcel file, checksum file and shift parcel files to cloudera parcel directory and change ownership to cloudera-scm user
+
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Remove new-line between comment and code

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+sudo tar zcvf ${filen^^}-$version-$OS.parcel ${filen^^}-$version/ --owner=root --group=root
+sudo sha1sum ${filen^^}-$version-$OS.parcel | awk '{ print $1 }' > ${filen^^}-$version-$OS.parcel.sha
+
+if [[ "$placefile" == "true" ]];
+	then
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Place `then` on same line as if statement.
https://google.github.io/styleguide/shell.xml#Loops

File: examples/cloudera-parcel-gcsconnector/create_parcel.sh [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+	sudo cp ${filen^^}-$version-$OS.parcel* /opt/cloudera/parcel-repo/ ; sudo chown cloudera-scm:cloudera-scm /opt/cloudera/parcel-repo/*
```
Comment (ryanmcdowell) [inline examples/cloudera-parcel-gcsconnector/create_parcel.sh]: Separate statements to different lines.

Line-length of shell files should be kept to 80 chars.
https://google.github.io/styleguide/shell.xml?showone=Line_Length_and_Long_Strings#Line_Length_and_Long_Strings

**Comment by yhqs540** [comment]
Hi Ryan, the comments were fixed

**Pull request merged**